handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -734,7 +734,10 @@ private void wrapNonAppData(ChannelHandlerContext ctx, boolean inUnwrap) throws
             // See https://github.com/netty/netty/issues/5860
             while (!ctx.isRemoved()) {
                 if (out == null) {
-                    out = allocateOutNetBuf(ctx, 0);
+                    // As this is called for the handshake we have no real idea how big the buffer needs to be.
+                    // That said 2048 should give us enough room to include everything like ALPN / NPN data.
+                    // If this is not enough we will increase the buffer in wrap(...).
+                    out = allocateOutNetBuf(ctx, 2048);
```

Comment:
mental note to my self ... we may be able to use something like [MAX_BIO_BYTES](https://github.com/netty/netty/pull/6201/files#diff-82f94494cc35616a8a19b6aa9ed033efR90) in the future or create another define for "non-app-data"